processor/transformprocessor/internal/metrics/func_extract_sum_metric_test.go

```diff
@@ -0,0 +1,336 @@
+				sumMetric.Sum().SetIsMonotonic(true)
+
+				sumMetric.SetName(histogramMetric.Name() + "_sum")
+				dp := sumMetric.Sum().DataPoints().AppendEmpty()
+				dp.SetDoubleValue(histogramMetric.Histogram().DataPoints().At(0).Sum())
+
+				attrs := getTestAttributes()
+				attrs.CopyTo(dp.Attributes())
+			},
+		},
+		{
+			name:         "histogram (cumulative)",
+			input:        getTestHistogramMetric(),
+			temporality:  pmetric.AggregationTemporalityCumulative,
+			monotonicity: false,
+			want: func(metrics pmetric.MetricSlice) {
+				histogramMetric := getTestHistogramMetric()
+				histogramMetric.CopyTo(metrics.AppendEmpty())
+				sumMetric := metrics.AppendEmpty()
+				sumMetric.SetEmptySum()
+				sumMetric.Sum().SetAggregationTemporality(pmetric.AggregationTemporalityCumulative)
+				sumMetric.Sum().SetIsMonotonic(false)
+
+				sumMetric.SetName(histogramMetric.Name() + "_sum")
+				dp := sumMetric.Sum().DataPoints().AppendEmpty()
+				dp.SetDoubleValue(histogramMetric.Histogram().DataPoints().At(0).Sum())
+
+				attrs := getTestAttributes()
+				attrs.CopyTo(dp.Attributes())
+			},
+		},
+		{
+			name:         "exponential histogram",
+			input:        getTestExponentialHistogramMetric(),
+			temporality:  pmetric.AggregationTemporalityDelta,
+			monotonicity: false,
+			want: func(metrics pmetric.MetricSlice) {
+				expHistogramMetric := getTestExponentialHistogramMetric()
+				expHistogramMetric.CopyTo(metrics.AppendEmpty())
+				sumMetric := metrics.AppendEmpty()
+				sumMetric.SetEmptySum()
+				sumMetric.Sum().SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+				sumMetric.Sum().SetIsMonotonic(false)
+
+				sumMetric.SetName(expHistogramMetric.Name() + "_sum")
+				dp := sumMetric.Sum().DataPoints().AppendEmpty()
+				dp.SetDoubleValue(expHistogramMetric.ExponentialHistogram().DataPoints().At(0).Sum())
+
+				attrs := getTestAttributes()
+				attrs.CopyTo(dp.Attributes())
+			},
+		},
+		{
+			name:         "exponential histogram (monotonic)",
+			input:        getTestExponentialHistogramMetric(),
+			temporality:  pmetric.AggregationTemporalityDelta,
+			monotonicity: true,
+			want: func(metrics pmetric.MetricSlice) {
+				expHistogramMetric := getTestExponentialHistogramMetric()
+				expHistogramMetric.CopyTo(metrics.AppendEmpty())
+				sumMetric := metrics.AppendEmpty()
+				sumMetric.SetEmptySum()
+				sumMetric.Sum().SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+				sumMetric.Sum().SetIsMonotonic(true)
+
+				sumMetric.SetName(expHistogramMetric.Name() + "_sum")
+				dp := sumMetric.Sum().DataPoints().AppendEmpty()
+				dp.SetDoubleValue(expHistogramMetric.ExponentialHistogram().DataPoints().At(0).Sum())
+
+				attrs := getTestAttributes()
+				attrs.CopyTo(dp.Attributes())
+			},
+		},
+		{
+			name:         "exponential histogram (cumulative)",
+			input:        getTestExponentialHistogramMetric(),
+			temporality:  pmetric.AggregationTemporalityCumulative,
+			monotonicity: false,
```

Comment:
Can you add a test for cumulative and monotonic? And if that isn't a valid combination, let's validate that in the function.